### ISSUE TITLE
5.3.4 fails to build with `mtl-2.3.1`

### ISSUE THREAD

**Issue by andreasabel** [mihaimaruseac/hindent#697]
On GHC 9.6:
```
$ cabal install hindent
...
Build profile: -w ghc-9.6.1 -O1
In order, the following will be built (use -v for more details):
 - hindent-5.3.4 (lib) (requires build)
 - path-io-1.8.0 (lib) (requires build)
 - hindent-5.3.4 (exe:hindent) (requires build)
...
[4 of 5] Compiling HIndent.Pretty   ( src/HIndent/Pretty.hs, dist/build/HIndent/Pretty.o, dist/build/HIndent/Pretty.dyn_o )

src/HIndent/Pretty.hs:67:12: error: [GHC-88464]
    Variable not in scope: when :: Bool -> Printer () -> Printer a2
   |
67 |            when (i == 0) newline
   |            ^^^^
...
```

**Comment by toku-sa-n** [comment]
Thank you for the report. The current latest is 6.0.0, and ghc 9.6.1 should be able to compile HIndent 6.0.0. Could you try it?

**Comment by andreasabel** [comment]
Looks like for GHC 9.6 you are blocked on `ghc-lib-parser`:
```
$ cabal install hindent-6.0.0
...
Error: cabal: Could not resolve dependencies:
[__0] trying: hindent-6.0.0 (user goal)
[__1] next goal: ghc-lib-parser (dependency of hindent)
[__1] rejecting: ghc-lib-parser-9.6.1.20230312 (conflict: hindent =>
ghc-lib-parser>=9.4.1.20220807 && <9.5)
[__1] trying: ghc-lib-parser-9.4.4.20221225
[__2] next goal: base (dependency of hindent)
[__2] rejecting: base-4.18.0.0/installed-4.18.0.0 (conflict: ghc-lib-parser =>
base>=4.15 && <4.18)
```

**Comment by andreasabel** [comment]
I revised `hindent < 6` with bound `mtl < 2.3` on Hackage, see e.g. https://hackage.haskell.org/package/hindent-5.3.4/revisions/
This helps those that use `cabal` to install `hindent`.

**Comment by toku-sa-n** [comment]
Oh, sorry, and thank you for the revision. I'll resolve the compilation of 6.0.0 with GHC 9.6 later.

**Comment by mihaimaruseac** [comment]
Thank you very much @andreasabel
